lib/endpoints/class-wp-json-post-types-controller.php

```diff
@@ -3,6 +3,33 @@
 class WP_JSON_Post_Types_Controller extends WP_JSON_Controller {
 
 	/**
+	 * Register the routes for the objects of the controller
+	 * 
+	 */
+	public function register_routes() {
+
+		parent::register_routes();
+
+		register_json_route( 'wp', '/types', array(
+			'methods'         => WP_JSON_Server::READABLE,
+			'callback'        => array( $this, 'get_items' ),
+			'args'            => array(
+				'post_type'          => array(),
+			),
+		) );
+
+		register_json_route( 'wp', '/types/schema', array(
+			'methods'         => WP_JSON_Server::READABLE,
+			'callback'        => array( $this, 'get_item_schema' ),
+		) );
+
+		register_json_route( 'wp', '/types/(?P<type>[\w-]+)', array(
+			'methods'         => WP_JSON_Server::READABLE,
+			'callback'        => array( $this, 'get_item' ),
+		) );
```

Comment:
I think it's fine as-is here, but we probably need a `deregister_json_route` as well.

One thing I did like from v1 was that you took the `$routes` array and returned your modified version (filter-style), which meant you could `unset` anything you didn't want. Don't know how that would work with `register_json_route` though.